utils/src/communication.rs

```diff
@@ -9,6 +9,28 @@ use std::{
 
 use crate::comp_decomp::BitPack;
 
+#[derive(PartialEq, Clone, Serialize, Deserialize, Debug)]
+pub enum Position {
+    Pixel(f32, f32),
+    Percent(f32, f32),
+}
```

Comment:
Why don't you make this store just a singular value? (just one `f32`). Then you could just send a tuple of `Position` and you could store a mix of percentage and absolute values